apps/minifront/src/state/swap/helpers.ts

```diff
@@ -63,22 +64,69 @@ export const sendCandlestickDataRequest = async (
     throw new Error('Asset pair equivalent');
   }
 
-  try {
-    const { data } = await dexClient.candlestickData(
+  const suppressAbort = (err: unknown) => {
+    if (signal && !signal.aborted) {
+      throw err;
+    }
+  };
+
+  const directReq = dexClient
+    .candlestickData(
       {
         pair: { start, end },
         limit,
       },
       { signal },
-    );
-    return data;
-  } catch (err) {
-    if (err instanceof Error && err.name === 'AbortError') {
-      return;
-    } else {
-      throw err;
+    )
+    .catch(suppressAbort);
+
+  const inverseReq = dexClient
+    .candlestickData(
+      {
+        pair: { start: end, end: start },
+        limit,
+      },
+      { signal },
+    )
+    .catch(suppressAbort);
+
+  const candlestickData = (await directReq)?.data ?? [];
+
+  const inverseCandlestickData = (await inverseReq)?.data ?? [];
```

Comment:
Can you add documentation about the purpose of querying the inverse? It's not clear from this code why it's necessary.